src/providers/details/tabs/Hosts/components/VSphereNetworkModal.tsx

```diff
@@ -0,0 +1,101 @@
+import { type FC, type ReactNode, useCallback, useState } from 'react';
+import { validateNoSpaces } from 'src/modules/Providers/utils/validators/common';
+import { useForkliftTranslation } from 'src/utils/i18n';
+
+import ModalForm from '@components/ModalForm/ModalForm';
+import type { NetworkAdapters, V1beta1Provider } from '@kubev2v/types';
+import { Form, ModalVariant } from '@patternfly/react-core';
+
+import { VSphereEndpointType } from './utils/constants';
+import { getSelectedInventoryHostNetworkTriples } from './utils/getSelectedInventoryHostNetworkTriples';
+import { onSaveHost } from './utils/onSaveHost';
+import type { InventoryHostNetworkTriple } from './utils/types';
+import HostsNetworksSelect from './HostsNetworksSelect';
+import HostsNetworksSetPassword from './HostsNetworksSetPassword';
+import HostsNetworksSetUserName from './HostsNetworksSetUserName';
+
+import './VSphereNetworkModal.style.css';
+
+type VSphereNetworkModalProps = {
+  provider: V1beta1Provider;
+  data: InventoryHostNetworkTriple[];
+  selectedIds: string[];
+};
+
+const VSphereNetworkModal: FC<VSphereNetworkModalProps> = ({ data, provider, selectedIds }) => {
+  const { t } = useForkliftTranslation();
+  const [alertMessage, setAlertMessage] = useState<ReactNode>(null);
```

Comment:
why its ReactNode? we talked aobut a string

Reply:
I removed this hook, no need since it's handled already by the ModalForm